src/coreclr/tools/aot/ILCompiler/RdXmlRootProvider.cs

```diff
@@ -153,14 +153,93 @@ private void ProcessMethodDirective(IRootingServiceProvider rootProvider, Module
+                        {
+                            return false;
+                        }
+
+                        MethodSignature pointerSignatureType = ((FunctionPointerType)type).Signature;
+                        MethodSignature pointerSignatureParam = ((FunctionPointerType)parameter).Signature;
+
+                        for (int i = 0; i < pointerSignatureType.Length; i++)
+                        {
+                            if (!InstantiationMatches(pointerSignatureType[i], pointerSignatureParam[i], instArgs))
+                            {
+                                return false;
+                            }
+                        }
+
+                        return InstantiationMatches(pointerSignatureType.ReturnType, pointerSignatureParam.ReturnType, instArgs);
+
+                    case TypeFlags.SignatureMethodVariable:
+                        SignatureMethodVariable sigMethodVar = (SignatureMethodVariable)type;
+                        return sigMethodVar.Index < instArgs.Count && instArgs[sigMethodVar.Index] == parameter;
+
+                    case TypeFlags.SignatureTypeVariable:
+                        SignatureTypeVariable sigTypeVar = (SignatureTypeVariable)type;
+                        return sigTypeVar.Index < instArgs.Count && instArgs[sigTypeVar.Index] == parameter;
+
+                    case TypeFlags.GenericParameter:
+                        throw new ArgumentException();
+
+                    default:
+                        if (parameter.Instantiation.Length != type.Instantiation.Length)
+                        {
+                            return false;
+                        }
+
+                        for (int i = 0; i < type.Instantiation.Length; i++)
+                        {
+                            if (type.Instantiation[i].ContainsSignatureVariables())
+                            {
+                                if (!InstantiationMatches(type.Instantiation[i], parameter.Instantiation[i], instArgs))
+                                {
+                                    return false;
+                                }
+                            }
+                            else if (type.Instantiation[i] != parameter.Instantiation[i])
+                            {
+                                return false;
+                            }
+                        }
+
+                        return true;
+                }
+            }
+
+            static bool SignatureMatches(MethodDesc method, List<TypeDesc> parameter, List<TypeDesc> instArgs)
             {
                 if (parameter.Count != method.Signature.Length)
+                {
                     return false;
+                }
 
                 for (int i = 0; i < method.Signature.Length; i++)
                 {
-                    if (method.Signature[i] != parameter[i])
+                    if (!InstantiationMatches(method.Signature[i], parameter[i], instArgs))
```

Comment:
Good trick with just having this match the instantiated siganture.

Wondering if we would get the same effect if we just move the `SignatureMatches` check after the `MakeInstantiatedMethod` - we would be comparing the already instantiated signature that way.

Reply:
I'll give it a try.

Reply:
I think they're the same.
Done in 4ea4b2f.